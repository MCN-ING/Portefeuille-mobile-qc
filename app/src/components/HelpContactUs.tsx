import { useTheme } from '@hyperledger/aries-bifold-core'
import { StyleSheet, Text, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'

import CalendarImg from '../assets/img/calendar-empty.svg'
import ClockImg from '../assets/img/clock.svg'
import PhoneImg from '../assets/img/telephone.svg'

type DetailsType = {
  daysOpen: string
  openingHours: string
  phone: string
  phoneSec?: string
}

type ContactType = {
  details: DetailsType
}

type HelpContactUsProps = {
  itemContact: ContactType[]
}
const HelpContactUs = ({ itemContact = [] }: HelpContactUsProps) => {
  const { ColorPallet, TextTheme } = useTheme()

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: ColorPallet.brand.primaryBackground,
    },

    textHeaderTitle: {
      ...TextTheme.headingThree,
      flexShrink: 1,
      color: TextTheme.headingThree.color,
      paddingTop: 8,
      paddingBottom: 8,
    },
    textSectionTitle: {
      ...TextTheme.title,
      flexShrink: 1,
      color: TextTheme.bold.color,
      paddingTop: 8,
      paddingBottom: 8,
    },
    section: {
      paddingVertical: 12,
    },
    sectionRow: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    sectionDoubleRow: {
      paddingTop: 10,
      flexDirection: 'row',
      alignItems: 'flex-start',
      height: 100,
    },
    phoneImage: {
      width: 24,
      height: 24,
    },
    sectionDescription: {
      ...TextTheme.normal,
      color: TextTheme.normal.color,
      textAlign: 'left',
      textDecorationLine: 'none',
      marginLeft: 10,
    },
  })

  return (
    <SafeAreaView style={styles.container} edges={['left', 'right']}>
      {itemContact.map((item, index) => (
        <>
          <View key={index}>
            <View style={styles.sectionRow}>
              <CalendarImg />
              <Text style={styles.sectionDescription}> {item.details.daysOpen}</Text>
            </View>
            <View style={styles.sectionRow}>
              <ClockImg />
              <Text style={styles.sectionDescription}> {item.details.openingHours}</Text>
            </View>
            <View style={styles.sectionDoubleRow}>
              <PhoneImg style={styles.phoneImage} />
              <Text style={styles.sectionDescription}>
                {item.details.phone}
                {'\n'}
                {item.details.phoneSec}
              </Text>
            </View>
          </View>
        </>
      ))}
    </SafeAreaView>
  )
}

export default HelpContactUs
