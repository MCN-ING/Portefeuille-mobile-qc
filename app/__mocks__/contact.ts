import { HelpContactUsProps } from '../src/components/HelpContactUs'

export const contactMock: HelpContactUsProps = {
  itemContact: [
    {
      details: {
        daysOpen: 'any_days_open',
        openingHours: 'any_opening_hours',
        phone: 'any_phone',
      },
    },
  ],
}
