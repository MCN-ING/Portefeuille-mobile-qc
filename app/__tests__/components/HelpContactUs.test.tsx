import { render } from '@testing-library/react-native'
import HelpContactUs from '../../src/components/HelpContactUs'
import { contactMock } from '../../__mocks__/contact'

describe('HelpContactUs Component', () => {
  test('Renders Correctly', () => {
    const tree = render(<HelpContactUs itemContact={contactMock.itemContact} />)

    expect(tree).toMatchSnapshot()
  })
})
