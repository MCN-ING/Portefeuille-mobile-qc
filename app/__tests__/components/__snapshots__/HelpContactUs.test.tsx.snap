// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpContactUs Component Renders Correctly 1`] = `
<RNCSafeAreaView
  edges={
    [
      "left",
      "right",
    ]
  }
  style={
    {
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      < />
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
            "marginLeft": 10,
            "textAlign": "left",
            "textDecorationLine": "none",
          }
        }
      >
         
        any_days_open
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      < />
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
            "marginLeft": 10,
            "textAlign": "left",
            "textDecorationLine": "none",
          }
        }
      >
         
        any_opening_hours
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "height": 100,
          "paddingTop": 10,
        }
      }
    >
      <
        style={
          {
            "height": 24,
            "width": 24,
          }
        }
      />
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
            "marginLeft": 10,
            "textAlign": "left",
            "textDecorationLine": "none",
          }
        }
      >
        any_phone
        

      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
